### ISSUE TITLE
Pressing the reboot made the NSPanel hang at bootup

### ISSUE THREAD

**Issue by rleongcs** [Blackymas/NSPanel_HA_Blueprint#1399]
Yesterday I have updated 2 of my NSPanel from 4.0.2 to the latest 4.1.4.

1. Initially I was met with compile errors and after increasing my VM memory from 2GB to 4GB, I was able to compile successfully and uploaded to the NSPanels.
2. I was also able to upload the latest 4.1.4 TFT with no issues.
3. The NSPanel during boot shows that the TFT is 4.1.4, ESPHome is 4.0.2 and Blueprint is blank and retrying (pic below).
4. I tried to do a "Clean Build Files".
5. Upload again.
6. This time the bootup screen shows TFT is 4.1.4, ESPHome is 4.1.4 and Blueprint is 4.1.4 and then moved on to the home screen.
7. Everything is working good from here.
8. Until I tested the "Reboot" button, the same issue again happens as in Item 3.
9. I have to repeat Item 5, then 6 and 7 is all good.

So basically now cannot reboot NSPanel else will have to upload again for it to work.
@edwardtfn any idea what might be the issue here? 

![IMG_2578](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/61628312/b3668469-89b2-403f-b907-f6ff0d89212e)

**Comment by rleongcs** [comment]
> Are you using framework esp-idf?

Yes. Switched from the old arduino.

> To solve that, either flash your device using serial, so the partitions will be rebuilt accordingly, or you can flash it via OTA using the original framework.

I will try it and revert.

**Comment by edwardtfn** [comment]
Hey @rleongcs, any update to this?

**Comment by edwardtfn** [comment]
I'm closing as we have no updates and I'm pretty confident the proposed change will solve the issue. Please feel free to reopen in if that is not the case.

**Comment by bbbenji** [comment]
Same here. I am on `ESPHome 2023.11.6`, but when rebooting, according to the logs at least, it reverts the FW back to `2023.10.4`. Then gets stuck on the screen shown in the OP.

Changing the framework type to `arduino` and reflashing via OTA seems to fix the issue.

Is there any way to rebuild the partitions according to esp-idf without serial flashing?

**Comment by edwardtfn** [comment]
Unfortunately I don't know another way to do this. 😞
